doc/specs/#476 - Pinned packages.md

```diff
@@ -0,0 +1,180 @@
+
+```
+Name                                                 Id                                Version    
+------------------------------------------------------------------------------------------------
+Microsoft Bob                                        Microsoft.Bob                     2.35.0
+iTunes                                               Apple.iTunes                      12.0.199.4
+
+```
+
+```
+winget pin --set <package> 
+```
+
+`<package>` correlates to a query, similar to `winget list`. When running this command, the package information will be stored in the table in the tracking catalog. This command should be able to accept Package Identifers (checked against all repositories) as well as names, tags, etc. If the package does not have a version number locally but is available from a source, then the version number from the source should be used.
+
+If there is no package directly available for software a user wants to pin, but it is [included as part of another package](https://github.com/microsoft/winget-cli/issues/1073), the package which includes said piece of software should be treated as pinned unless a newer version doesn't change the version of the pinned software. 
+
+```
+winget pin --clear <package>
+```
+
+It functions similarly to `winget pin --set`, but it clears the pin. This should automatically happen on `winget uninstall`, or if it is detected the package is no longer present on the system.
+
+Enhancements will also need to be made to other commands in the Windows Package Manager, including:
+
+```
+winget upgrade --include-pinned
+```
+
+With the new `--include-pinned` argument, Upgrade should show a separate table in addition to the regular table which shows possible upgrades to pinned packages.
+
+```
+<...>
+Upgrades to pinned packages: 
+
+Name                                                 Id                                Version       Available     Source
+-------------------------------------------------------------------------------------------------------------------------
+Clippy for Azure 2011 Datacenter                     Microsoft.Clippy.2011.Datacenter  2.35.0        2.35.1.2      winget
+```
+
+
+```
+winget <upgrade/install> --include-pinned <package>
+```
+
+Upgrade and Install will now need to use the same argument to bypass the pin for a package temporarily, in which case the pin is updated to pin the new version after the installation completes.
+
+```
+winget install --pin <package>
+```
+
+Install should also be extended to allow the user to pin a package at install time.
+
+
+
+
+In addition to commands, there should be a new `Pinned` key in exported package lists, allowing users to move their pins to different systems:
+
+```
+// ...
+    {
+      "Packages": 
+      [
+        {
+          "PackageIdentifier": "Microsoft.Bob",
+          "PackageVersion" : 2.35.0,
+          "Pinned" : "true"
+        },
+      ],
+    }
+// ...
+```
```

Comment:
Shouldn't the key data type be boolean?

Reply:
Yes, yes it should. I wasn't 100% awake when I was working on this.

Reply:
Resolved!